.travis.sh

```diff
@@ -1,13 +1,14 @@
 #!/bin/sh
 
-# Install libgit2 from a bunch of Ubuntu 15.04 binary repos
-mkdir libgit2
-pushd libgit2
-
-wget -q --progress=bar "https://launchpad.net/ubuntu/+source/libgit2/0.21.1-1/+build/6494190/+files/libgit2-21_0.21.1-1_amd64.deb"
-wget -q --progress=bar "https://launchpad.net/ubuntu/+source/libgit2/0.21.1-1/+build/6494190/+files/libgit2-dev_0.21.1-1_amd64.deb"
-
-sudo dpkg -i "libgit2-21_0.21.1-1_amd64.deb"
-sudo dpkg -i "libgit2-dev_0.21.1-1_amd64.deb"
-
-popd
+# Install libgit2 from source and pip-install pygit2
+echo "Installing libgit2"
+wget https://github.com/libgit2/libgit2/archive/v0.22.0.tar.gz
+tar xzf v0.22.0.tar.gz
+cd libgit2-0.22.0/
+cmake -Wno-dev .
+make --quiet
+sudo make install --quiet
+sudo ldconfig
+sudo pip install pygit2
+status=$( python -c "import pygit2;print('SUCCESS')" )
+echo "$status"
```

Comment:
Your original was better; since python indeed sets the exitcode to a non-zero value when an exception was uncaught (I thought we would need to explicitly call sys.exit(1)).

My bad :)